backend/src/main/java/fk/prof/backend/model/assignment/WorkAssignmentScheduleFactory.java

```diff
@@ -0,0 +1,38 @@
+package fk.prof.backend.model.assignment;
+
+import recording.Recorder;
+
+public class WorkAssignmentScheduleFactory {
+  private final int windowDurationInMins;
+  private final int windowEndToleranceInSecs;
+  private final int schedulingBufferInSecs;
+  private final int minAcceptableDelayForWorkAssignmentInSecs;
+  private final int maxAcceptableDelayForWorkAssignmentInSecs;
+
+  public WorkAssignmentScheduleFactory(int windowDurationInMins,
+                                       int windowEndToleranceInSecs,
+                                       int schedulingBufferInSecs,
+                                       int maxAcceptableDelayForWorkAssignmentInSecs)
+      throws IllegalArgumentException {
+    this.windowDurationInMins = windowDurationInMins;
+    this.windowEndToleranceInSecs = windowEndToleranceInSecs;
+    this.schedulingBufferInSecs = schedulingBufferInSecs;
+    this.maxAcceptableDelayForWorkAssignmentInSecs = maxAcceptableDelayForWorkAssignmentInSecs;
+    this.minAcceptableDelayForWorkAssignmentInSecs = schedulingBufferInSecs / 2;
+
+    if(this.maxAcceptableDelayForWorkAssignmentInSecs < (this.minAcceptableDelayForWorkAssignmentInSecs * 2)) {
+      throw new IllegalArgumentException(String.format("Max acceptable delay for work assignment = %d" +
+          "should be at least be twice of min acceptable delay for work assignment = %d", maxAcceptableDelayForWorkAssignmentInSecs, minAcceptableDelayForWorkAssignmentInSecs));
+    }
+  }
+
+  public WorkAssignmentSchedule getNewWorkAssignmentSchedule(Recorder.WorkAssignment.Builder[] workAssignmentBuilders,
```

Comment:
Added this purely for readability purpose (as a later stage refactoring) since the args accepted here are always passed through to work assignment schedule and do not change in the process lifetime, so encapsulating them in this class.
I can revert back to the existing behavior, just makes constructor definitions for aggregation window planner and work assignment schedule longer by 5 params everywhere.
Maybe having a final object which has these params as members and passing that to work assignment schedule is another way